File: classic/quine.c
/*
 * C Quine - 经典实现
 * 
 * 编译和验证:
 *     gcc -o quine quine.c && ./quine | diff - quine.c
 */

#include <stdio.h>

int main() {
    char *s = "#include <stdio.h>%c%cint main() {%c    char *s = %c%s%c;%c    printf(s, 10, 10, 10, 34, s, 34, 10, 10, 10);%c    return 0;%c}%c";
    printf(s, 10, 10, 10, 34, s, 34, 10, 10, 10);
    return 0;
}

File: classic/quine.java
/*
 * Java Quine - 经典实现
 * 
 * 编译和验证:
 *     javac Quine.java && java Quine | diff - Quine.java
 */

public class Quine {
    public static void main(String[] args) {
        String s = "public class Quine {\n    public static void main(String[] args) {\n        String s = %c%s%c;\n        System.out.printf(s, 34, s, 34);\n    }\n}\n";
        System.out.printf(s, 34, s, 34);
    }
}
